docs/command reference.md

```diff
@@ -0,0 +1,151 @@
+psabpf-ctl multicast-group del-member pipe ID MULTICAST_GROUP egress-port OUTPUT_PORT instance INSTANCE_ID
+
+MULTICAST_GROUP := id MULTICAST_GROUP_ID
+```
+
+# Pipelines and ports management
+
+```shell
+psabpf-ctl pipeline load id ID PATH
+psabpf-ctl pipeline unload id ID
+psabpf-ctl add-port pipe id ID dev DEV
+psabpf-ctl del-port pipe id ID dev DEV
+```
+
+# Tables
+
+```shell
+psabpf-ctl table add pipe ID TABLE ACTION key MATCH_KEY [data ACTION_PARAMS] [priority PRIORITY]
+psabpf-ctl table add pipe ID TABLE ref key MATCH_KEY data ACTION_REFS [priority PRIORITY]
+psabpf-ctl table update pipe ID TABLE ACTION key MATCH_KEY [data ACTION_PARAMS] [priority PRIORITY]
+psabpf-ctl table delete pipe ID TABLE [key MATCH_KEY]
+psabpf-ctl table default set pipe ID TABLE ACTION [data ACTION_PARAMS]
+psabpf-ctl table get pipe ID TABLE [ref] [key MATCH_KEY]
+
+TABLE := { id TABLE_ID | name FILE | TABLE_FILE }
+ACTION := { id ACTION_ID | ACTION_NAME }
+ACTION_REFS := { MEMBER_REF | group GROUP_REF } 
+MATCH_KEY := { EXACT_KEY | LPM_KEY | RANGE_KEY | TERNARY_KEY | none }
+EXACT_KEY := { DATA }
+LPM_KEY := { DATA/PREFIX_LEN }
+RANGE_KEY := { DATA_MIN..DATA_MAX }
+TERNARY_KEY := { DATA^MASK }
+ACTION_PARAMS := { DATA | counter COUNTER_NAME COUNTER_VALUE | meter METER_NAME METER_VALUE }
+COUNTER_VALUE := { BYTES | PACKETS | BYTES:PACKETS }
+METER_VALUE := { PIR:PBS CIR:CBS }
+```
+
+Commands to implement:
+```shell
+psabpf-ctl table default get pipe ID TABLE
+```
+
+`ref` keyword means that table has an implementation, `ActionProfile` or `ActionSelector`, and then behave according to
+this situation.
+
+# Action Selectors
+
+```shell
+psabpf-ctl action-selector add_member pipe ID ACTION_SELECTOR ACTION [data ACTION_PARAMS]
+psabpf-ctl action-selector delete_member pipe ID ACTION_SELECTOR MEMBER_REF
+psabpf-ctl action-selector update_member pipe ID ACTION_SELECTOR MEMBER_REF ACTION [data ACTION_PARAMS]
+psabpf-ctl action-selector create_group pipe ID ACTION_SELECTOR
+psabpf-ctl action-selector delete_group pipe ID ACTION_SELECTOR GROUP_REF
+psabpf-ctl action-selector add_to_group pipe ID ACTION_SELECTOR MEMBER_REF to GROUP_REF
+psabpf-ctl action-selector delete_from_group pipe ID ACTION_SELECTOR MEMBER_REF from GROUP_REF
+psabpf-ctl action-selector default_group_action pipe ID ACTION_SELECTOR ACTION [data ACTION_PARAMS]
+
+ACTION_SELECTOR := { id ACTION_SELECTOR_ID | name FILE | ACTION_SELECTOR_FILE }
+ACTION := { id ACTION_ID | ACTION_NAME }
+ACTION_PARAMS := { DATA }
+```
+
+# Meters
+
+```shell
+psabpf-ctl meter get pipe ID METER index INDEX
+psabpf-ctl meter update pipe ID METER index INDEX PIR:PBS CIR:CBS
+psabpf-ctl meter reset pipe ID METER index INDEX
+
+METER := { id METER_ID | name FILE | METER_FILE }
+INDEX := { DATA }
+PIR := { DATA }
+PBS := { DATA }
+CIR := { DATA }
+CBS := { DATA }
+```
+
+# Digests
+
+```shell
+psabpf-ctl digest get pipe ID DIGEST
+
+DIGEST := { id DIGEST_ID | name FILE | DIGEST_FILE }
```

Comment:
It'd be good to explain what is `DIGEST_ID` and `DIGEST_FILE`. This comment applies to all other commands too. 

Also, it would be useful to explain what is the name format as a P4 developer may try to use extern name defined as defined in a P4 language while we require `<Pipeline>_<extern name>`, right? As this applies to all externs/objects, we can add such a note at the beginning of this document.